common/lib/common-utils/src/test/mocha/idleTaskScheduler.spec.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { SinonFakeTimers, useFakeTimers, fake, match } from "sinon";
+import * as idleTask from "../../idleTaskScheduler";
+
+describe("Idle task scheduler", () => {
+    let clock: SinonFakeTimers;
+
+    beforeEach(() => {
+        clock = useFakeTimers();
+    });
+
+    afterEach(() => {
+        clock.reset();
+    });
+
+    after(() => {
+        clock.restore();
+    });
+
+    function someTask(x: number): boolean {
+        return x > 3 ? true : false;
+    }
+
+    it("Should schedule and run a synchronous task during idle time", async () => {
+        const requestIdleCallbackMock = fake((callback, timeout) => {
+            callback(timeout);
+        });
+        (globalThis as any).requestIdleCallback = requestIdleCallbackMock;
+
+        await idleTask.scheduleIdleTask(() => {
+            someTask(5);
+        }, 1000);
+
+        assert(requestIdleCallbackMock.calledOnce);
+        assert(requestIdleCallbackMock.calledWith(match.func, { timeout: 1000 }));
+    });
+
+    it("Should fall back to setTimeout when idle Task API is not available", async () => {
+        let success = false;
+        await new Promise((resolve, reject) => {
```

Comment:
Might be worth asserting here that (globalThis as any).requestIdleCallback === undefined.
That assert essentially validates that we are running code in Node (not browser) and that two UTs are covering all cases.
If it were to change in the future (i.e. we start running tests in browser), it would catch that and force developers to re-evaluate test strategy (i.e. add UT that would explicitly remove globalThis.requestIdleCallback to test timer case)